common/djangoapps/third_party_auth/saml.py

```diff
@@ -84,6 +85,17 @@ def generate_saml_config(self, idp=None):
         else:
             return super().generate_saml_config()
 
+    def auth_complete(self, *args, **kwargs):
+        """
+        Handle exceptions that happen during SAML authentication
+        """
+        try:
+            super().auth_complete(*args, **kwargs)
+        # We are seeing errors of MultiValueDictKeyError looking for the parameter 'RelayState'.
+        # We would like to have a more specific error to handle for observability purposes.
+        except MultiValueDictKeyError as e:
+            raise AuthMissingParameter(e)
```

Comment:
1. Would it be useful to log this? That might be a way to both ignore the error, and alert on a really high volume by alerting on the logs.
2. Are you intending to ignore the new error? Other?

Also, I'd love to get @johnnagro or someone from Enterprise to review, since they are the ones that are mostly helping customers with SAML issues, and I know John was looking into surfacing SAML errors to the end user.

Reply:
This actually turns this error into the same error as the other error, which is happening in Google OAuth, but I was vaguely thinking of trying to tune something for both of them with your help. I am open to other suggestions of how to handle it, but I figured I would take a first stab so we have something to talk about.

Reply:
Ok.

Also, does AuthMissingParameter take the missing parameter, and is there a way to pull it from MultiValueDictKeyError?

Reply:
Yup, I just ran into that now, and I fixed it so that it does have the correct version of the parameter.